test/Microsoft.Azure.Devices.Edge.Test/Context.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test
+{
+    using System;
+    using System.IO;
+    using System.Net;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Configuration;
+
+    public class Context
+    {
+        static readonly Lazy<Context> Default = new Lazy<Context>(
+            () =>
+            {
+                IConfiguration context = new ConfigurationBuilder()
```

Comment:
should it be simple to have only 1 config source? all parameters are coming from a single json file.

Reply:
I'm not sure what you're recommending here...?

Reply:
current implementation takes context.json plus environment variable with "E2E_" prefix.  What are "E2E_" enviornment variables? can these variables define in context.json?

Reply:
Ah, I see. 😃 The E2E_* values can be defined in context.json--the ConfigurationBuilder will take them from the config file _or_ the environment. But the E2E_* values specifically are secrets, and we'd rather not have them saved to disk. So I made sure that the tests would still run if they were defined in the environment instead of the config file. In automated builds this should integrate nicely with secrets management.

Reply:
would you please give me an example of E2E_* variable?

Reply:
The E2E_* variables required for the tests so far are:
```
E2E_CONTAINER_REGISTRY_PASSWORD
E2E_EVENT_HUB_ENDPOINT
E2E_IOT_HUB_CONNECTION_STRING
```

Reply:
thanks.  Does this document down in config file or somewhere?

Reply:
It's not documented yet. I'll add something later when things settle.